va/va.go

```diff
@@ -487,7 +498,10 @@ func (va *ValidationAuthorityImpl) checkCAAService(ctx context.Context, ident co
 		if prob.Type == probs.ServerInternalProblem {
 			prob.Detail = "Internal communication failure"
 		} else {
-			prob.Detail = err.Error()
+			// bgrpc.CodeToProblem can return non-gRPC errors passed back by
```

Comment:
So this clause is meant to handle non-gRPC errors? Why are we using `grpc.ErrorDesc` then? Why not just `err.Error()`? Also, maybe it makes more sense for `bgrpc.CodeToProblem` to return a `ProblemDetails` rather than a `Type`?